src/analysis/retail/priest/shadow/modules/spells/ShadowWordDeath.tsx

```diff
@@ -53,15 +57,21 @@ class ShadowWordDeath extends ExecuteHelper {
   }
 
   adjustMaxCasts() {
+    //The spellusable of SW:D is being tracked properly, however the value of the casts is not correct.
+    //I do not know why that is the case.
+    //To fix this, we calcuate the number of casts of SW:D
+    //It is equal to executeTime/10s plus the number of procs of SW:D from Death and Madness, plus the number of procs from DeathSpeaker
     const cooldown =
       this.abilities.getAbility(TALENTS.SHADOW_WORD_DEATH_TALENT.id)!.cooldown * 1000;
     const ExecuteCasts = Math.ceil(this.totalExecuteDuration / cooldown);
 
-    if (this.selectedCombatant.hasTalent(TALENTS.DEATH_AND_MADNESS_TALENT)) {
-      //The Death and Madness talent lets you cast twice each cast in execute.
-      this.maxCasts += ExecuteCasts;
-    }
-    this.maxCasts += ExecuteCasts + this.totalNonExecuteCasts;
+    const DeathAndMadnessCasts = this.deathandmadness.getResets();
+    const DeathSpeakerCasts = this.deathspeaker.getProcsUsed();
+
+    //console.log("SWD Totals:",ExecuteCasts,"NE",this.totalNonExecuteCasts,"DM",DeathAndMadnessCasts,"DS",DeathSpeakerCasts);
+
+    this.maxCasts +=
```

Comment:
should this be `this.maxCasts =` instead of `+=`? i don't think this method *should be* called twice but not sure its guaranteed not to be

Reply:
It should be.  I've fixed this.